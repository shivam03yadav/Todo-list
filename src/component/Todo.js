import React, { Component } from 'react'

class Todo extends Component {

    constructor(props) {
        super(props)
    
        this.state = {
             input: '',
             items: []
        }
    }

    update = (event) => {
        this.setState({
            input: event.target.value
        })
    }
    add = () => {
        this.setState(prev => {
             return{
                 input: '',
                 items: prev.items.concat(prev.input)
             }
        })
    }

    deleteTodo = (key) => {
        this.setState(prev => {
            let items = [...prev.items];

            items.splice(key, 1);

            return{
                input: '',
                items
            };
        })

    }
    
    render() {
        const { items, input } = this.state
        return (
            <div>
                <h1> todo list</h1>
                <h4>Count is {items.length}</h4>
                <input 
                     className="input"
                     type="text" 
                     value={input} 
                     onChange={this.update}/>
                <button 
                     id="addBtn"
                     type="button" 
                     onClick={this.add}>add item</button>
                <ul>
                {
                    items.map((item, index) => (
                        <div>
                        <li key={index}>{item}{ ' '}
                        <button onClick={() => this.deleteTodo(index)}>X</button>
                        </li>
                        </div>
                    ))
                }
                </ul>
            </div>
        )
    }
}

export default Todo
