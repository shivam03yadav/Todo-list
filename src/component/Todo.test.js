import React from 'react'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Todo from './Todo';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter()});

describe("todo testing", () => {
    let wrapper;
    beforeEach(() => {
         wrapper = shallow( <Todo /> );
    });

    it('renders correctly when there are no items', () => {
        const tree = renderer.create(<Todo />).toJSON();
        expect(tree).toMatchSnapshot();
      });

    it("should render the wrapper", () => {
        expect(wrapper).toBeDefined();
    })
    it("should call the functions", () => {
        const { add, update, deleteTodo } = wrapper.instance();
        add();
        update( {target: { value: "hello" } })
        deleteTodo( {key: "123"});
    })
    it("should check the input value", () => {
        wrapper.find("input").simulate("change", {
            target: { value: "hello" }
        })
    })
    it('should render the heading', () => {
        expect(wrapper.find("h1").text()).toContain("todo list");
    })
    it('should render the add button to check initial click', () => {
        wrapper.find("#addBtn").simulate("click");
        const text = wrapper.find("items")
        expect(text.length).toBe(0);
    })
    
})