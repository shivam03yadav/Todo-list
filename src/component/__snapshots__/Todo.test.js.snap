// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo testing renders correctly when there are no items 1`] = `
<div>
  <h1>
     todo list
  </h1>
  <h4>
    Count is 
    0
  </h4>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    id="addBtn"
    onClick={[Function]}
    type="button"
  >
    add item
  </button>
  <ul />
</div>
`;
